File: app/src/main/java/co/edu/uniautonoma/posgradosapp/Actividades/BaseActivity.java
package co.edu.uniautonoma.posgradosapp.Actividades;

import android.app.ProgressDialog;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {

    private ProgressDialog dialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        dialog = new ProgressDialog(this);
        dialog.setMessage("Cargando...");
        dialog.setCancelable(true);
        dialog.setIndeterminate(true);
    }

    protected void mostrarDialog() {

        if(!dialog.isShowing()){
            dialog.show();
        }
    }

    protected void ocultarDialog() {

        if(dialog.isShowing()) {
            dialog.dismiss();
        }
    }
}

File: presentation/src/main/java/co/edu/uniautonoma/posgradosapp/presentation/ui/base/BaseActivity.kt
package co.edu.uniautonoma.posgradosapp.presentation.ui.base

import android.app.ProgressDialog
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity() {

    private var dialog: ProgressDialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        dialog = ProgressDialog(this)
        dialog!!.setMessage("Cargando...")
        dialog!!.setCancelable(true)
        dialog!!.isIndeterminate = true
    }

    protected fun mostrarDialog() {
        if (!dialog!!.isShowing) {
            dialog!!.show()
        }
    }

    protected fun ocultarDialog() {
        if (dialog!!.isShowing) {
            dialog!!.dismiss()
        }
    }
}